llvm/lib/Transforms/Vectorize/SLPVectorizer.cpp

```diff
@@ -10341,7 +10470,7 @@ class BoUpSLP::ShuffleInstructionBuilder final : public BaseShuffleAnalysis {
       InVectors.push_back(V1);
   }
   /// Adds another one input vector and the mask for the shuffling.
-  void add(Value *V1, ArrayRef<int> Mask) {
+  void add(Value *V1, ArrayRef<int> Mask, bool = false) {
```

Comment:
What is the point of this bool argument?

Reply:
It is a unification with the cost estimator interface, check `void add(Value *V1, ArrayRef<int> Mask, bool ForExtracts = false)` from ShuffleCostEstimator. Same functions with the same args will be used for both code gen/cost estimation.